packages/backend/src/auth-service-client/client.ts

```diff
@@ -0,0 +1,88 @@
+interface Tenant {
+  id: string
+  idpConsentUrl: string
+  idpSecret: string
+}
+
+export class AuthServiceClientError extends Error {
+  constructor(
+    message: string,
+    public status: number,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public details?: any
+  ) {
+    super(message)
+    this.status = status
+    this.details = details
+  }
+}
+
+export class AuthServiceClient {
+  private baseUrl: string
+
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private async request<T = any>(
+    path: string,
+    options: RequestInit
+  ): Promise<T> {
+    const response = await fetch(`${this.baseUrl}${path}`, options)
+
+    if (!response.ok) {
+      let errorDetails
+      try {
+        errorDetails = await response.json()
+      } catch {
+        errorDetails = { message: response.statusText }
+      }
+
+      throw new AuthServiceClientError(
+        `Auth Service Client Error: ${response.status} ${response.statusText}`,
+        response.status,
+        errorDetails
+      )
+    }
+
+    if (
+      response.status === 204 ||
+      response.headers.get('Content-Length') === '0'
+    ) {
+      return undefined as T
+    }
+
+    const contentType = response.headers.get('Content-Type')
+    if (contentType && contentType.includes('application/json')) {
+      try {
+        return (await response.json()) as T
+      } catch (error) {
+        throw new AuthServiceClientError(
+          `Failed to parse JSON response from ${path}`,
+          response.status
+        )
+      }
+    }
+
+    return (await response.text()) as T
+  }
+
+  public tenant = {
+    get: (id: string) =>
+      this.request<Tenant>(`/tenant/${id}`, { method: 'GET' }),
+    create: (data: Omit<Tenant, 'id'>) =>
+      this.request('/tenant', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(data)
+      }),
+    update: (id: string, data: Partial<Omit<Tenant, 'id'>>) =>
+      this.request(`/tenant/${id}`, {
+        method: 'PATCH',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(data)
+      }),
+    delete: (id: string) => this.request(`/tenant/${id}`, { method: 'DELETE' })
+  }
```

Comment:
could probably move these methods and the `Tenant` interface out of this file if we want but opted to just put it here since it's really the only thing we need for the forseeable future.